id/id.go

```diff
@@ -0,0 +1,20 @@
+// Package id contains facilities for generating and validating random IDs
+// which are meant to be used as uploaded filenames.
+package id
+
+import (
+	"crypto/rand"
+	"math"
+	"math/big"
+)
+
+// Generate a random ID like "1njfizqgeukrq", "3l44ze7pf47fd", "35dgd4n5ryup", etc.
+func Generate() (id string, err error) {
```

Comment:
Package `id` is created specifically for InstantShare needs and not meant to be a general purpose one. At least at this time, since InstantShare is the only intended user. We could even make it an internal package to enforce no one else can import it, but, I don't think it's needed because no one will anyway. Making packages internal has a cost of making the import path slightly more verbose.

So, that's why I chose to make the exported API of the package as simple as possible. If InstantShare needs to change, say, length of IDs or characters used, we can modify this package directly.

If InstantShare needs to to support variable ID length, etc., then we can make it general.

The principle we follow here is to avoid prematurely creating unnecessary abstractions. Instead, we write the simplest code that is actually needed today. This is pretty commonly done in Go projects.